v2/spiffetls/dial_test.go

```diff
@@ -192,58 +210,58 @@ func TestDialWithMode(t *testing.T) {
 	for _, test := range tests {
 		test := test
 
-		err := os.Setenv("SPIFFE_ENDPOINT_SOCKET", test.defaultWlAPIAddr)
-		require.NoError(t, err)
+		if test.defaultWlAPIAddr != "" {
+			err := os.Setenv("SPIFFE_ENDPOINT_SOCKET", test.defaultWlAPIAddr)
+			require.NoError(t, err)
+		}
 
 		t.Run(test.name, func(t *testing.T) {
 			// Start listening
 			listenCtx, cancelListenCtx := context.WithTimeout(context.Background(), 10*time.Second)
 			defer cancelListenCtx()
 
+			var wg sync.WaitGroup
```

Comment:
If you defer this wait then you don't have to do the `loop` label below and can just return instead of `break loop`:
```suggestion
			var wg sync.WaitGroup
			defer wg.Wait()
```